app/assets/v2/js/shared.js

```diff
@@ -159,10 +159,16 @@ var sanitizeDict = function(d, keyToIgnore) {
 };
 
 var sanitizeAPIResults = function(results, keyToIgnore) {
-  for (var i = 0; i < results.length; i++) {
-    results[i] = sanitizeDict(results[i], keyToIgnore);
+  if (results.length >= 1) {
+    for (let i = 0; i < results.length; i++) {
+      results[i] = sanitizeDict(results[i], keyToIgnore);
+    }
+    return results;
   }
-  return results;
+
+  results = [results];
```

Comment:
whats up with the list wrapping here?

Reply:
For proper operation in all cases and backward compatibility, it was necessary to make an array/object also for individual data entries (e.g. issue title from github).

Then there is a reference to `sanitizeDict` function which checks if the input is an object and only if it is an object removes unwanted characters from them. Someone has previously created an oversight, and therefore downloaded issues titles from GitHub or other data that were not in the array were not cleaned and skipped and could be abused.

`sanitizeDict` creates a loop that cleans every element from the array one by one, therefore it was necessary to replace elements without arrays (i.e. single ones) into arrays to keep the function and backward compatibility and at the same time make single data also be cleaned.

Reply:
So this code is to support the case where the `results` argument passed in is not an array. But on line 169 you wrap `resutls` into an array, but then call `santizeDict()` on the first element in the array-wrapped `results` object, which is the same as what it was before it was wrapped, right? I guess I'm just asking whats the difference between the code in this PR lines 169-171 and something like:

`return sanitizeDict(results, keyToIgnore);`

Reply:
Jeez, you're absolutely right! I didn't even notice that I had too much of it and used a few phrases when you can do all the same in one line you mentioned xD Corrected in the latest commit :D